app/models/field.rb

```diff
@@ -6,12 +6,18 @@ class Field < ApplicationRecord
   
   validates :category, presence: true
   validates :rank, presence: true, uniqueness: { scope: :category }
-  validates :field_type, inclusion: { in: %i[boolean choice] }
+  validates :field_type, inclusion: { in: %w[boolean choice] }
   
   serialize :choices, Array
   validates :choices, presence: true, if: :multiple_choice?
-  
+
+  scope :in_category, ->(category) { where category: category }
+
   def multiple_choice?
     field_type == :choice
   end
+
+  def self.categories
+    order(:category).pluck(:category).uniq
+  end
```

Comment:
These methods are only used from the bus stop model, but I still prefer keeping some of the field-related functionality in its own model file.

Reply:
I agree with that preference.